webapp/components/test-runs.html

```diff
@@ -65,8 +66,13 @@
           }
           // Fetch by products.
           if (this.productSpecs && this.productSpecs.length) {
-            runs.push(fetch(`/api/runs${this.query}`)
-              .then(r => r.ok && r.json()));
+            runs.push(
+              fetch(`/api/runs${this.query}`)
+                .then(r => r.ok && r.json().then(runs => {
```

Comment:
I saw your earlier change #758 which throws an exception if fetch fails. Shall we modify this (and `loadMoreRuns`) to throw/reject if `!r.ok` by the way?

Reply:
No; a 404 is legitimate since the query can be user-input. We combine things in the `Promise.all` below, and I wouldn't want multiple sources to cancel each other because one 404s.